internal/toolsig/path.go

```
@@ -0,0 +1,268 @@
+	if len(segs) == 0 || args == nil {
+		return
+	}
+	// The root is always an object: MCP tool arguments are a JSON object.
+	if segs[0].isIdx {
+		return
+	}
+
+	// container holds whatever we are currently writing into, addressed through
+	// the parent so we can replace it when it turns out to be the wrong type.
+	var (
+		curMap map[string]any = args
```

golangci-lint (2.12.2)

[error] ST1023: should omit type map[string]any from declaration; it will be inferred from the right-hand side

(staticcheck)